app/js/arethusa.search/services/search.js

```diff
@@ -229,46 +229,44 @@ angular.module('arethusa.search').service('search', [
       var combine = 0;
       // latin enclytics usually are preceded with a '-' and may be
       // either right after the base word or shifted to right before it
-      if (!match && wordANext && wordANext.match(/^-/)) {
-        match = compareWords(wordA + wordANext.replace(/^-/,''),wordB);
+      if (!match && wordANext) {
+        var cleanWordANext = wordANext.replace(/^-/,'').replace(/-$/,'');
```

Comment:
Since `replace(/^-/,'').replace(/-$/,'');` is used a number of times, it might be beneficial to move it into a function. That way if the regex ever changes, it only needs to be changed in one place.

Reply:
good idea. I'll do that before merging. thanks!